src/getTreeColumnWidth.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  Cube,
+  DataVisualizationWidgetMapping,
+  LevelCoordinates,
+  getHierarchy,
+  getLevel,
+} from "@activeviam/activeui-sdk";
+import { quote } from "@activeviam/mdx";
+
+/**
+ * Returns the column width for the tree table calculated by the baseWidth + (maxLevelDepth * levelMultiplier)
+ */
+export function getTreeColumnWidth({
+  maxLevelDepth,
+  mapping,
+  treeTableColumnWidth,
+  columnLevels,
+  cube,
+}: {
+  mapping: DataVisualizationWidgetMapping;
+  maxLevelDepth: number;
+  treeTableColumnWidth: [number, number];
+  columnLevels: LevelCoordinates[];
```

Comment:
This should be called `rowLevels` (same remark as above). I also believe that this information is redundant with `mapping.rows` and that we should remove one of these two arguments.

If `rowLevels` contains the measures, then I believe that using it is simpler. But if it does not contain this information, then we should use `mapping.rows`.

I might be wrong, but if we base ourselves on `rowLevels`, then I believe that the function boils down to:

```
if (rowLevels.length === 0) {
  return undefined;
}

const { dimensionName, hierarchyName } = rowLevels[0];

if (dimensionName === "Measures") {
  return "[Measures].[Measures]";
}

const hierarchy = getHierarchy({ dimensionName, hierarchyName });
const firstLevel = hierarchy.levels[hierarchy.slicing ? 0 : 1];
return quote(dimensionName, hierarchyName, firstLevel.name);
```